plugins/net_plugin/include/eosio/net_plugin/net_utils.hpp

```diff
@@ -51,18 +51,19 @@ namespace detail {
    }
 
    /// @return host, port, remainder
-   inline std::tuple<std::string, std::string, std::string> split_host_port_remainder(const std::string& peer_add, bool should_throw) {
+   inline std::tuple<std::string, std::string, std::string> split_host_port_remainder(const std::string& peer_add_input, bool should_throw) {
       using std::string;
       // host:port[:trx|:blk][:<rate>]
-      if (peer_add.empty()) {
-         EOS_ASSERT(!should_throw, chain::plugin_config_exception, "Address specification is empty" );
+      if (peer_add_input.size() > max_p2p_address_length) {
```

Comment:
Why not trimming before this?

Reply:
Because the string could potentially be very large (larger than max_p2p_address_length).

Reply:
I think if a string after trimmed is less than the max size, it should be allowed.

Reply:
This is not a big concern.